action.yml

```diff
@@ -115,25 +127,26 @@ runs:
     - name: Updating bootstrap
       shell: bash
       run: |
+        echo -e 'Types: deb\nURIs:${{ inputs.mirror }}\nSuites: ${{ inputs.suites }}\nComponents: ${{ inputs.components }}\n' | sudo tee /opt/procursus/etc/apt/sources.list.d/procursus.sources
         sudo apt-get -y update
-        sudo apt-get -y dist-upgrade || :
+        sudo apt-get -y --allow-downgrades dist-upgrade || :
     
-    - name: Caching bootstrap
-      if: steps.information.outputs.procursus_bootstrapped != 'true' && inputs.cache == 'true' 
-      shell: bash
-      run: |
-        sudo mkdir -p ${{ inputs.cache-path }}
-        sudo rsync -alHh --inplace /opt/procursus ${{ inputs.cache-path }}
-        sudo rm -rf ${{ inputs.cache-path }}/procursus/var/cache/apt/archives/partial \
-          ${{ inputs.cache-path }}/procursus/var/lib/apt/lists/partial \
-          ${{ inputs.cache-path }}/procursus/Library/dpkg/triggers/Lock
-
     - name: Install packages
       shell: bash
       if: inputs.packages
       run: |
         sudo apt-get install -y ${{ inputs.packages }}
         
+    - uses: webiny/action-post-run@2.0.1
```

Comment:
Caching before installing packages was intentional, as I wanted the bootstrapped environment to be fresh no matter if it was bootstrapped normally or restored from cache. 

Also, caching after a job has finished will also cache unwanted modifications and potentially cause problems (who knows what happens to `/opt/procursus` after this step finishes?). CI is ideally supposed to have the same state at the start of each run.